### PR TITLE
reuse input trees vector in Resolver::resolveSigs

### PR DESCRIPTION
**Pull request by froydnj**
### Motivation

Per the PR title.  This saves some memory churn, since we can reuse the allocation of `trees`.  (We also weren't reserving space in `combinedTrees`, either.)  Notice that we already do the same thing in ResolveConstantsWalk.

### Test plan

See included automated tests.

### CODE REVIEW

**Comment by froydnj** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_KYmAoqsoRB4IPw
→ https://go/builds/bui_KYmAYiKShYOlpM

File: resolver/resolver.cc [hunk 1]
```diff
@@ -3144,7 +3145,7 @@ ast::ParsedFilesOrCancelled Resolver::resolveSigs(core::GlobalState &gs, vector<
         }
     }
 
-    return move(combinedTrees);
+    return move(trees);
```
Comment (elliottt) [inline resolver/resolver.cc:3148]: Is the move necessary here?

**Review by elliottt — APPROVED** [review]
Looks great, good call on avoiding the reallocation 👍

File: resolver/resolver.cc (hunk 1 above)
Comment (froydnj) [reply resolver/resolver.cc]: Nope, fixed!

**Pull request merged**